server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3092,8 +3092,12 @@ public Vlan createVlanAndPublicIpRange(final long zoneId, final long networkId,
             final String guestNetworkCidr = zone.getGuestNetworkCidr();
             if (guestNetworkCidr != null) {
                 if (NetUtils.isNetworksOverlap(newCidr, guestNetworkCidr)) {
-                    throw new InvalidParameterValueException("The new IP range you have specified has  overlapped with the guest network in zone: " + zone.getName()
-                            + ". Please specify a different gateway/netmask.");
+                    // when adding shared network with same cidr of zone guest cidr,
+                    // if the specified vlan is not present in zone, physical network, allow to create the network as the isolation is based on VLAN.
+                    if (_zoneDao.findVnet(zoneId, physicalNetworkId, vlanId).size() > 0) {
```

Comment:
@jburwell : Thanks for your suggestion. I will consider this change. The effect would be same if we put all the conditions with && operator.